data/validateQuestionLink.js

```diff
@@ -0,0 +1,80 @@
+import { questions } from "./question-data.js";
+import fs from "fs/promises";
+import path from "path";
+
+// Run this script and check logs/broken-links-logs.csv for broken links
+
+const STATUS_SUCCESS = 200;
+const LOGS_PATH = "../logs/broken-links-logs.csv";
+
+// The input param is the link to be validated on.
+export const validateURLResponse = async (input) => {
+  try {
+    const response = await fetch(input);
+
+    const status = response.status;
+
+    return status === STATUS_SUCCESS;
+  } catch (error) {
+    console.error(
+      `An error occured while accessing "${input}". \nError: ${error}`,
+    );
+    return false;
+  }
+};
+
+const validateQuizLink = async (linkList) => {
+  console.info("validating links...", linkList.length);
+  const badLinks = [];
+
+  await Promise.all(
+    linkList.map(async ({ link }, idx) => {
+      const result = await validateURLResponse(link);
+      if (!result) {
+        console.error("Found invalid link:", link, "at index", idx);
+        badLinks.push({ link });
+      }
+    }),
+  );
+  return badLinks;
+};
+
+async function logError(errorLog, message) {
+  await fs.appendFile(errorLog, message + "\n");
+}
+
+/**
+ * Sometimes multiple retries in validation of links is required since there
```

Comment:
Are they all running asynchronously? Can we not make it run synchronously.

Reply:
Unsure about the question. The `badLinks` is dependent on the previous operation where the previous `badLinks` were checked. Feel free to give me an example

Reply:
I was just wondering what was the reason behind running it few times (As in if there is a function running asynchronously where if you run it in a few attempts, it's all good).